src/librustdoc/html/templates/sidebar.html

```diff
@@ -0,0 +1,32 @@
+{%- if it.is_struct()
+        || it.is_trait()
+        || it.is_primitive()
+        || it.is_union()
+        || it.is_enum()
+        || it.is_mod()
+        || it.is_typedef() -%}
+<h2 class="location"><a href="#">
+{%- match it.kind.as_ref() -%}
+{%- when clean::ModuleItem with (_) -%}
+{%- if it.is_crate() %}Crate{% else %}Module{% endif %} {#+ -#}
+{%- else -%}
+{%- endmatch -%}
+{{- it.name.as_ref().unwrap() -}}
+</a></h2>{% endif -%}
+<div class="sidebar-elems">
+{%- if it.is_crate() -%}
+<ul class="block">
+{%- match cx.cache().crate_version -%}
+{% when Some with (version) %}<li class="version">Version {{version}}</li>
+{%- else -%}
+{% endmatch -%}
+<li><a id="all-types" href="all.html">All Items</a></li>{# -#}
+</ul>
+{%- endif %}
```

Comment:
As another example, this could be simplified with the addition of two fields to `Sidebar`:

```
is_crate_page: bool,
crate_version: &'str, // only present on crate pages
```

Then the template would become:

```
{%- if is_crate_page -%}
<ul class="block"> {#- -#}
  <li class="version">Version {{version}}</li> {#- -#}
  <li><a id="all-types" href="all.html">All Items</a></li> {#- -#}
</ul> {#- -#}
{%- endif -%}
```

By the way, note that I use the double-sided whitespace-eliminating comment `{#- -#}` instead of the single-sided on `{# -#}`. That allows the comment to stand off from the HTML by a space, which I think is a little more readable.

The single-sided comment `{# -#}` is needed when an HTML tag with attributes is spread over multiple lines, in order to preserve one space between attributes. Otherwise the attributes would get jammed together.